src/main/java/mil/dds/anet/threads/mart/services/MartReportImporterService.java

```diff
@@ -0,0 +1,371 @@
+        tasks.add(task);
+      } else {
+        errors.add(String.format("Can not find task: '%s' with uuid: %s", value, key));
+      }
+    });
+    return tasks;
+  }
+
+  private boolean assertAllowedMimeType(final String mimeType) {
+    return AttachmentResource.getAllowedMimeTypes().contains(mimeType);
+  }
+
+  private ReportDto getReportInfo(String reportJson) {
+    ReportDto report = null;
+    try {
+      report = ignoringMapper.readValue(reportJson, ReportDto.class);
+      if (report.getSubmittedAt() == null) {
+        logger.warn("Submitted time not provided in report");
+      } else {
+        final Duration transportDelay = Duration.between(report.getSubmittedAt(), Instant.now());
+        logger.info("Time between report send and reception: {} ms", transportDelay.toMillis());
+      }
+      logger.debug("Found report with UUID={}", report.getUuid());
+    } catch (JsonParseException jsonParseException) {
+      logger.error("e-mail does not look like JSON", jsonParseException);
+    } catch (JsonMappingException jsonMappingException) {
+      logger.error("Invalid JSON format", jsonMappingException);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    return report;
+  }
+
+  private void processReportInfo(ReportDto martReport, MartImportedReport martImportedReport,
+      List<FileAttachment> attachments) {
+    List<String> errors = new ArrayList<>();
+
+    // Validate author organization
+    final Organization organization = organizationDao.getByUuid(martReport.getOrganizationUuid());
+    if (organization == null) {
+      errors.add(String.format("Can not find submitter organization: '%s' with uuid: %s",
+          martReport.getOrganizationName(), martReport.getOrganizationUuid()));
+    }
+
+    // Validate report location
+    final Location location = locationDao.getByUuid(martReport.getLocationUuid());
+    if (location == null) {
+      errors.add(String.format("Can not find report location: '%s' with uuid: %s",
+          martReport.getLocationName(), martReport.getLocationUuid()));
+    }
+
+    if (organization != null && location != null) {
+      // Move on with the report
+      Report anetReport = new Report();
+      // Location
+      anetReport.setLocation(location);
+      List<ReportPerson> reportPeople = handleReportPeople(martReport, organization, errors);
+      // Report people
+      anetReport.setReportPeople(reportPeople);
+      // Report generic details
+      anetReport.setUuid(martReport.getUuid());
+      anetReport.setCreatedAt(martReport.getCreatedAt());
+      anetReport.setIntent(martReport.getIntent());
+      if (martReport.getAtmosphere() != null) {
+        anetReport
+            .setAtmosphere(Report.Atmosphere.valueOf(martReport.getAtmosphere().toUpperCase()));
+      }
+      anetReport.setEngagementDate(martReport.getEngagementDate());
+      anetReport.setReportText(martReport.getReportText());
+      anetReport.setClassification("NKU");
```

Comment:
This should not be hard-coded. Instead it should come from the report (there's a custom field called `securityMarking` that should contain the correct value). Site admins need to make sure that both sides have this marking in their configuration.